src/types/UIProtocol.ts

```diff
@@ -13,13 +13,23 @@ export enum ProtocolVersion {
   '0.0.1' = '0.0.1',
 }
 
-export enum ProtocolCommand {
+export enum MessageCode {
+  REQUEST = 0,
+  ANSWER = 1,
+  ERROR = 2,
+}
+
+export type ProtocolMessage = [MessageCode, ...ProtocolRequest];
+
+export type ProtocolRequest = [number, ...ProtocolCommand];
+
+export enum CommandCode {
```

Comment:
It's really the procedure name to run on the other WS endpoint. Not sure the type name change makes it clearer.